examples/HelloWorldExampleDS/HelloWorldSubscriber.cpp

```diff
@@ -126,34 +139,35 @@ bool HelloWorldSubscriber::init(Locator_t server_address)
 
 HelloWorldSubscriber::~HelloWorldSubscriber()
 {
-    Domain::removeParticipant(mp_participant);
+    DomainParticipantFactory::get_instance()->delete_participant(mp_participant);
 }
 
-void HelloWorldSubscriber::SubListener::onSubscriptionMatched(
-    Subscriber* /*sub*/,
-    MatchingInfo& info)
+void HelloWorldSubscriber::SubListener::on_subscription_matched(
+        DataReader* /*sub*/,
+        const SubscriptionMatchedStatus& info)
 {
-    if (info.status == MATCHED_MATCHING)
+    if (info.current_count_change == 1)
     {
-        n_matched++;
+        n_matched = info.total_count;
```

Comment:
Same comment as in the publisher. I think we should keep the internal count or use instead `current_count` instead of `total_count`. Although maybe we can just remove the variable because it is not being used.